tests/unit/ProductFilter/Options/AttributeOptionsProviderTest.php

```diff
@@ -0,0 +1,133 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace ProductFilter\Options;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\Module\PsxMarketingWithGoogle\ProductFilter\Options\AttributeOptionsProvider;
+use PrestaShop\Module\PsxMarketingWithGoogle\Repository\AttributesRepository;
+use PrestaShop\Module\PsxMarketingWithGoogle\Repository\LanguageRepository;
+
+class AttributeOptionsProviderTest extends TestCase
+{
+    public function testTransformRawDataToValidOptions()
+    {
+        $stubAttributeRepository = $this->createStub(AttributesRepository::class);
+        $stubLanguageRepository = $this->createStub(LanguageRepository::class);
+
+        $stubAttributeRepository->method('getCustomAttributesWithLocalizedValues')
+            ->willReturn($this->getInputData());
+        $stubLanguageRepository->method('getIsoById')
+            ->will($this->onConsecutiveCalls('fr', 'en_gb', 'fr', 'en_gb', 'fr', 'en_gb', 'fr', 'en_gb'));
+
+        $provider = new AttributeOptionsProvider(
+            $stubAttributeRepository,
+            $stubLanguageRepository
+        );
+        $result = $provider->getOptions();
+        $this->assertSame(
+            [
+                [
+                    'id' => '2',
+                    'key' => 'Form',
+                    'values' => [
+                        ['id' => '4', 'key' => 'Form', 'value' => 'Classique', 'language' => 'fr'],
+                        ['id' => '4', 'key' => 'Form', 'value' => 'Classic', 'language' => 'en_gb'],
+                        ['id' => '5', 'key' => 'Form', 'value' => 'Chromatique', 'language' => 'fr'],
+                        ['id' => '5', 'key' => 'Form', 'value' => 'Shiny', 'language' => 'en_gb'],
+                    ],
+                ],
+                [
+                    'id' => '1',
+                    'key' => 'Gender',
+                    'values' => [
+                        ['id' => '13', 'key' => 'Gender', 'value' => 'Femmes', 'language' => 'fr'],
+                        ['id' => '13', 'key' => 'Gender', 'value' => 'Women', 'language' => 'en_gb'],
+                        ['id' => '15', 'key' => 'Gender', 'value' => 'Hommes', 'language' => 'fr'],
+                        ['id' => '15', 'key' => 'Gender', 'value' => 'Men', 'language' => 'en_gb'],
+                    ],
+                ],
+            ],
+            $result
+        );
+    }
+
+    private function getInputData(): array
+    {
+        return [
+            [
+              'id_attribute_group' => '2',
+              'id_lang' => '1',
+              'name' => 'Classique',
+              'attribute_group' => 'Form',
+              'id_attribute' => '4',
+            ],
```

Comment:
the naming here seems dodgy.
we have "name", which is the attribute name.
we have the id_attribute and id_attribute_group.
But we have attribute_group while we should have group_name to be coherent